src/pynxtools/dataconverter/validation.py

```diff
@@ -847,6 +859,93 @@ def startswith_with_variations(
         # default
         return (False, 0)
 
+    def check_reserved_suffix(key: str, mapping: MutableMapping[str, Any]) -> bool:
+        """Check if an associated field exists for a key with a reserved suffix."""
+        reserved_suffixes = (
+            "_end",
+            "_increment_set",
+            "_errors",
+            "_indices",
+            "_mask",
+            "_set",
+            "_weights",
+            "_scaling_factor",
+            "_offset",
+        )
+
+        if not key.endswith((reserved_suffixes)):
+            # Ignore this test
+            return False
+
+        for suffix in reserved_suffixes:
+            if key.endswith(suffix) and key.rsplit(suffix, 1)[0] not in mapping:
+                collector.collect_and_log(
+                    key,
+                    ValidationProblem.ReservedSuffixWithoutField,
+                    key.rsplit(suffix, 1)[0],
+                )
+                return False
+        return True
+
+    def check_reserved_prefix(
+        key: str,
+        mapping: MutableMapping[str, Any],
+        nx_type: Literal["group", "field", "attribute"],
+    ) -> bool:
+        """Check if a reserved prefix was used in the correct context."""
+        reserved_prefixes = {
+            "attribute": {
+                "@BLUESKY_": None,  # do not use anywhere
+                "@DECTRIS_": "NXmx",
+                "@IDF_": None,  # do not use anywhere
+                "@NDAttr": None,
+                "@NX_": "all",
+                "@PDBX_": None,  # do not use anywhere
+                "@SAS_": "NXcanSAS",
+                "@SILX_": None,  # do not use anywhere
+            },
+            "field": {
+                "DECTRIS_": "NXmx",
+            },
+        }
+
+        prefixes = reserved_prefixes[nx_type]
+
+        if not key.rsplit("/", 1)[-1].startswith(tuple(prefixes.keys())):
+            # Ignore this test
+            return False
+
+        for prefix, context in prefixes.items():
+            if not key.rsplit("/", 1)[-1].startswith(prefix):
+                continue
+            if not context:
+                # This prefix should not be used by pynxtools.
+                collector.collect_and_log(
+                    prefix,
+                    ValidationProblem.ReservedPrefixInWrongApplication,
+                    None,
+                    key,
+                )
+                return False
+            elif context == "all":
+                # We can freely use this prefix everywhere.
+                continue
```

Comment:
Can't we return already here? We only end up here if the key starts with the prefix, so we probably don't have to check others.